GitUI/Script/ScriptRunner.cs

```diff
@@ -31,50 +31,53 @@ public static CommandStatus ExecuteScriptCommand(IWin32Window owner, GitModule m
 
         public static CommandStatus RunScript(IWin32Window owner, IGitModule module, string scriptKey, IGitUICommands uiCommands, RevisionGridControl revisionGrid)
         {
-            return RunScript(owner, module, scriptKey, uiCommands, revisionGrid,
-                msg => MessageBox.Show(owner, msg, Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error));
+            try
+            {
+                return RunScript(owner, module, scriptKey, uiCommands, revisionGrid,
+                    msg => MessageBox.Show(owner, msg, Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error));
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(owner,
+                    $"Failed to execute '{scriptKey}' script.{Environment.NewLine}Reason: {ex.Message}",
+                    Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
This should re-use a nearly identical new method of `MessageBoxes` from another open PR. (can be done in a follow-up)

Reply:
I wanted to keep it as small as possible due to plans to push it into 3.4.2.
Agree on a centralised message handling point, as well as pushing the strings out to be localisable. I plan to do this in the master for 4.0.